test/e2e/addon/kubestatemetrics.go

```diff
@@ -0,0 +1,186 @@
+package addon
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/eks"
+	"github.com/go-logr/logr"
+	appsv1 "k8s.io/api/apps/v1"
+	clientgo "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	ik8s "github.com/aws/eks-hybrid/internal/kubernetes"
+	"github.com/aws/eks-hybrid/internal/retry"
+	"github.com/aws/eks-hybrid/test/e2e/kubernetes"
+)
+
+const (
+	kubeStateMetricsName        = "kube-state-metrics"
+	kubeStateMetricsNamespace   = "kube-state-metrics"
+	kubeStateMetricsWaitTimeout = 2 * time.Minute
+)
+
+// KubeStateMetricsTest tests the kube-state-metrics addon
+type KubeStateMetricsTest struct {
+	Cluster   string
+	addon     *Addon
+	K8S       clientgo.Interface
+	EKSClient *eks.Client
+	K8SConfig *rest.Config
+	Logger    logr.Logger
+}
+
+// Create installs the kube-state-metrics addon
+func (k *KubeStateMetricsTest) Create(ctx context.Context) error {
+	k.addon = &Addon{
+		Cluster:   k.Cluster,
+		Namespace: kubeStateMetricsNamespace,
+		Name:      kubeStateMetricsName,
+	}
+
+	if err := k.addon.CreateAndWaitForActive(ctx, k.EKSClient, k.K8S, k.Logger); err != nil {
+		return err
+	}
+
+	if _, err := ik8s.GetAndWait(ctx, kubeStateMetricsWaitTimeout, k.K8S.AppsV1().Deployments(k.addon.Namespace), k.addon.Name, func(d *appsv1.Deployment) bool {
```

Comment:
you this a few times now, it may worth just adding it as a helper to kubernetes.DeploymentWaitForReplicas or something.